core/src/replay_stage.rs

```diff
@@ -3440,8 +3440,16 @@ impl ReplayStage {
         // 3) The best "selected" bank is on a different fork,
         //    switch_threshold succeeds
         let mut failure_reasons = vec![];
-        let selected_fork = {
-            let switch_fork_decision = tower.check_switch_threshold(
+        struct CandidateVoteAndResetBanks<'a> {
+            // A bank that the validator will consider voting on given it passes all
+            // remaining vote checks
+            candidate_vote_bank: Option<&'a Arc<Bank>>,
+            // A bank that the validator will reset its PoH to
```

Comment:
nit: will reset its PoH to whether or not it's voting ?

Reply:
changed